Format Python according to PEP 8

emileferreira opened the pull request:
I've formatted all the Python files in accordance with [PEP 8](https://peps.python.org/pep-0008/)—using space indentation, limiting lines to 79 characters, _et cetera_. This is in preparation for further, meaningful changes to the codebase. I've also removed unused imports and variables, except those in functions decorated as `UnfinishedFunc`.

emileferreira commented:
I've also noticed some undefined variables in the formatted `FunctionalSupport.py` at lines 257 and 267, and in `Evaluator.py` at lines 227, 504 and 556. We should take a look at resolving those.

gailweiss commented:
this is a very big change (touches all of the files), and anyway, i rather like tabs for indentation :) so i'm avoiding this one. That said, I won't object to receiving a separate PR for removing unused imports!

emileferreira commented:
Yes it is a very big change, which required a lot of time and effort, but I have ensured that all tests pass. A well-formatted codebase would make it easier for others to contribute to the project. I have now reverted the files to tab-based indentation and hope that you consider re-opening the pull request.

gailweiss commented:
Thanks! Can you please move the is/== change you made in Sugar.py to a separate PR, as it touches the code itself and I would like to handle it separately. Also, have you committed this with —ignore-rev, and if not can you add it (https://akrabat.com/ignoring-revisions-with-git-blame/)?

emileferreira commented:
I recall making such a change, but I don't see it in `Sugar.py`. Could you please point me to the line number? I have granted you write access to the pull request branch, should you wish to make any changes.

The `--ignore-rev` argument can't be used when making commits; it's only for use with `git blame` to locally ignore commits. Once you are satisfied with the changes, you can squash the commits to make you the author or I can edit the author details.

emileferreira commented:
I've reverted the offending commit and created a `.gitignore-revs` file. As per the linked article, you need to point to this file from your local Git config to automatically ignore the specified commits.

emileferreira commented:
I now realise why I couldn't see the change in `Sugar.py`; the original file used the `==` operator. I changed it to an `is` operator in 309fb81 before restoring it in 31ca5b2. I have reverted `Sugar.py` to the original operator.

gailweiss commented:
i went over it. had to make a few changes -- mostly fixing broken comments, but also some variable names -- and made a PR to your PR, i think. thanks for the thorough work, looking forward to merge once you merge in my changes

emileferreira commented:
Thanks for your changes—they've been merged. I've reverted your reversion of the ambiguous variable names, as discussed and in compliance with PEP 8's [names to avoid](https://peps.python.org/pep-0008/#names-to-avoid).

gailweiss commented on RASP_support/Support.py:
```diff
@@ -176,15 +176,17 @@ def zipmap(n, k_vars, func):
-			default, tuple), "aggregates on functions with single output \
-			should have scalar default"
+		errnote = "aggregates on functions with single output should have" \
+			+ "scalar default"
```
oops, can you add a space here to sit between "have" and "scalar" in the string?

emileferreira replied:
No problem :)

gailweiss commented on RASP_support/Evaluator.py:
```diff
@@ -224,7 +224,7 @@ def _evaluateDictComp(self, ast):
 			raise RASPTypeError(
 				"dict comprehension should have got a list or dict to loop "
-				+ "over, but got:", l)
+				+ "over, but got:", d)
```
good catch but i actually was hoping to get this in a separate pr cause it fixes an actual bug

gailweiss commented on RASP_support/Evaluator.py:
```diff
@@ -500,7 +500,7 @@ def _evaluateAggregateExpr(self, ast):
 
 		if not isinstance(sel, UnfinishedSelect):
-			raise RASPTypeError("Expected selector, got:", strdesc(selector))
+			raise RASPTypeError("Expected selector, got:", strdesc(sel))
```
please put in separate bug pr (not a pep 8 fix)

gailweiss commented on RASP_support/Evaluator.py:
```diff
@@ -552,7 +552,7 @@ def _evaluateApplication(self, ast, unf):
 			raise RASPTypeError("Applying unfinished expects to apply",
 								ENCODER_NAME, "or selector, got:",
-								strdesc(sel))
+								strdesc(unf))
```
please put in separate bug pr (not a pep 8 fix)

gailweiss reviewed (changes requested):
you fixed some bugs, which i appreciate but would prefer in a separate pr

gailweiss commented:
make sure to add the latest commits to the git blame ignore file (although maybe we'll squash and ill have to update it, ill see what happens - terrible at this)
